### PR TITLE
Reland "Simplify timestampWrites API" with the rest of the spec updated

### PR DESCRIPTION
**Pull request by kainino0x**
First commit reverts #3977 (relands #3930).
Remaining spec fixes are in the second commit.

### CODE REVIEW

**Comment by kainino0x** [comment]
@toji bump just in case this one got lost (no rush)

File: spec/index.bs [hunk 1]
```diff
@@ -12706,13 +12723,18 @@ Issue: Because timestamp query provides high-resolution GPU timestamp, we need t
...
-    - No two entries in |timestampWrites| may have the same `location`.
-    - For each |timestampWrite| in |timestampWrites|:
-        - |timestampWrite|.`querySet` must be [$valid to use with$] |device|.
-        - |timestampWrite|.`querySet`.{{GPUQuerySet/type}} is {{GPUQueryType/"timestamp"}}.
-        - |timestampWrite|.`queryIndex` &lt; |timestampWrite|.`querySet`.{{GPUQuerySet/count}}.
+    - {{GPUFeatureName/"timestamp-query"}} must be [=enabled for=] |device|.
+    - |timestampWrites|.`querySet` must be [$valid to use with$] |device|.
+    - |timestampWrites|.`querySet`.{{GPUQuerySet/type}} must be {{GPUQueryType/"timestamp"}}.
+    - |timestampWrites|.`beginningOfPassWriteIndex` must be &lt; |timestampWrite|.`querySet`.{{GPUQuerySet/count}}.
+    - |timestampWrites|.`endOfPassWriteIndex` must be &lt; |timestampWrite|.`querySet`.{{GPUQuerySet/count}}.
```
Comment (toji) [inline spec/index.bs:12730]: I'm guessing this is the text that the editorial comment means when it says some text may assume the value is undefined if it doesn't exist? Still feels a bit weird to says `undefined` is less than `count`.

Is it worth updating these statements to read something like "`beginningOfPassWriteIndex` must be `undefined` or < `count`"?

**Review by toji — APPROVED** [review]
LGTM with one nit.

**Comment by toji** [comment]
> @toji bump just in case this review got lost (no rush)

One month later: No rushing was done. 😅 (Sorry about that!)

File: spec/index.bs (hunk 1 above)
Comment (kainino0x) [reply spec/index.bs:12895]: er, no, it's other places in the spec that I didn't want to update right now. This is just a mistake, nice catch.

**Pull request merged**